ocaml/xapi/xapi_pool_update.ml

```diff
@@ -376,3 +376,18 @@ let detach_attached_updates __context =
     ignore(Helpers.call_api_functions ~__context
       (fun rpc session_id -> Client.Pool_update.detach ~rpc ~session_id ~self ~host))
   )
+
+let pool_update_download_handler (req: Request.t) s _ =
+  debug "pool_update.pool_update_download_handler URL %s" req.Request.uri;
+  let path =
+    if String.startswith Constants.get_pool_update_download_uri req.Request.uri then
```

Comment:
If we don't trust (the entity that supplied) the `req` parameter, then should also change the `else` clause so that it raises an exception instead.

With the code as it stands, `req` can be a straightforward `/path/to/some/file` and the file will be served.